eCommerce platforms/Adobe Magento/src/code/local/Microsoft/Dfp/controllers/Checkout/OnepageController.php

```diff
@@ -0,0 +1,399 @@
+		$payload = $this->generateBankEventPayload($type);
+		return $this->dfp->sendBankEvent($payload, $correlationId);
+	}
+
+	private function GetCardType($data)
+	{
+		$type = '';
+		switch ($data) {
+			case 'VI':
+				$type = 'Visa';
+				break;
+			case 'AE':
+				$type = 'Amex';
+				break;
+			case 'MC':
+				$type = 'Mastercard';
+				break;
+			case 'DI':
+				$type = 'Discover';
+				break;
+		}
+		return $type;
+	}
+
+	private function GetPaymentInstrumentDetails($order)
+	{
+		$PaymentData = $order->getPayment()->getData();
+
+		$billingAddress = $order->getBillingAddress();
+		$billingData = $billingAddress->getData();
+		$billingStreet = $billingAddress->getStreet();
+
+		$paymentInstrumentList = array();
+		$userCountryCode = 'USA';
+		$userCountryDialCode = '1';
+
+		$billingAddress = array(
+			"firstName"	     => $billingData['firstname'],
+			"lastName"	     => $billingData['lastname'],
+			"phoneNumber"    => "+" . $userCountryDialCode . "-" . $billingAddress->telephone,
+			"street1"	     => $billingStreet[0],
+			"street2"	     => count($billingStreet) > 1 ? $billingStreet[1] : null,
+			"city"		     => $billingAddress->city,
+			"state"		     => $billingAddress->region,
+			"zipCode"	     => $billingAddress->postcode,
+			"country"	     => $userCountryCode
+		);
+
+		switch ($PaymentData['method']) {
+			case 'ccsave':
+				$expirationDate = new DateTime($PaymentData['cc_exp_year'] . '-' . $PaymentData['cc_exp_month']);
+				$expirationDateISO =  $expirationDate->format(DateTime::ATOM);
+				$paymentInstrument = array(
+					'type'			  => 'CreditCard',
+					'cardType'		  => $this->GetCardType($PaymentData['cc_type']),
+					'holderName'	  => $PaymentData['cc_owner'],
+					'expirationDate'  => $expirationDateISO,
+					'lastFourDigits'  => $PaymentData['cc_last4'],
+					'purchaseAmount'  => $order->getGrandTotal(),
+					'billingAddress'  => $billingAddress
+				);
+				$paymentInstrumentList[] = $paymentInstrument;
+				break;
+			case 'checkmo':
+				$paymentInstrument = array(
+					'type'			 => 'Invoice',
+					'purchaseAmount' => $order->getGrandTotal(),
+					'billingAddress' => $billingAddress
+				);
+				$paymentInstrumentList[] = $paymentInstrument;
+				break;
+		}
+		return $paymentInstrumentList;
+	}
+
+	private function generatePurchasePayload($order, $assessmentType)
+	{
+		$billingAddress = $order->getBillingAddress();
+
+		$quote = Mage::getModel('sales/quote')->load($order->getQuoteId());
+		$userId = $order->customer_email;
+		if ($quote->getCheckoutMethod(true) == 'guest') {
+			$userId = $this->dfp->GUID();
+		}
+
+		// User Details
+		$userCountryCode = 'USA'; //this is something needs to set based on zip code, here we just hardcode a possible one
+		$userCountryDialCode = '1'; //this is something needs to set based on zip code, here we just hardcode a possible one
+		$timeZone = (Mage::getSingleton('core/session')->getCustomerTimeZone()) / 60;
+		$timeZone = ($timeZone > 0 ? '-' : '+') . $timeZone;
```

Comment:
0 is -? Normally see ± or nothing for 00:00

Reply:
Good point.  Will leave 0 with no prefix.